cmd/release-controller/http_candidate.go

```
@@ -0,0 +1,336 @@
+	latestPromotedTime := make(map[string]int64)
+	type stableRef struct {
+		from string
+		name string
+		time int64
+	}
+	stableReleases := make([]stableRef, 0)
+	stable, err := c.stableReleases(false)
+	if err != nil {
+		return releaseCandidates, err
+	}
+	for _, r := range stable.Releases {
+		for _, tag := range r.Release.Source.Spec.Tags {
+			if tag.Annotations[releaseAnnotationSource] != fmt.Sprintf("%s/%s", r.Release.Source.Namespace, r.Release.Source.Name) {
+				continue
+			}
+			if _, err := semverParseTolerant(tag.Name); err == nil {
+				t, _ := time.Parse(time.RFC3339, tag.Annotations[releaseAnnotationCreationTimestamp])
+				stableReleases = append(stableReleases, stableRef{from: tag.From.Name, name: tag.Name, time: t.Unix()})
+			}
+		}
+	}
+	sort.Slice(stableReleases, func(i, j int) bool {
+		vi, _ := semverParseTolerant(stableReleases[i].name)
+		vj, _ := semverParseTolerant(stableReleases[j].name)
+		return vi.GT(vj)
+	})
+	remaining := len(releaseStreams)
+	for _, r := range stableReleases {
+		if remaining == 0 {
+			break
+		}
+		v, _ := semverParseTolerant(r.name)
+
+		// Check if the stable version's <MAJOR>.<MINOR> matches any release stream that we are processing
+		found := false
+		for _, stream := range releaseStreams {
+			streamVersion, _ := semverParseTolerant(stream)
+			if next[stream] == nil && streamVersion.Major == v.Major && streamVersion.Minor == v.Minor {
+				found = true
+				break
+			}
+		}
+		if !found {
+			continue
+		}
+
+		// Call oc adm release info to get previous nightly info for the stable release
+		op, err := c.releaseInfo.ReleaseInfo(r.from)
+		if err != nil {
+			glog.Errorf("Could not get release info for tag %s: %v", r.from, err)
+			continue
+		}
+		releaseInfo := ReleaseInfoShort{}
+		if err := json.Unmarshal([]byte(op), &releaseInfo); err != nil {
+			glog.Errorf("Could not unmarshal release info for tag %s: %v", r.from, err)
```

this should return an error out of this method